Add xlayers

cspencerjones opened the pull request:
I know it's not used a lot, but I hope that with more testing and work, xlayers can become a regularly-used tool (and in order to do more testing I need to be able to run it on workers).

cspencerjones committed:
Add xlayers

I know it's not used a lot, but I hope that with more testing and work, xlayers can become a regularly-used tool (and in order to do more testing I need to be able to run it on workers).

rabernat commented:
Can someone re-trigger the circle-ci build. It just timed out.
https://circleci.com/gh/pangeo-data/pangeo-cloud-federation/981?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

I don't have permissions. Who does? @jhamman? @scottyhq?

rabernat commented:
Nevermind, I just needed to connect my github account to my circleci account.

rabernat commented:
Ok so building this image keeps timing out on the "Solving environment" step. I feel like we've been here before. What can we do?

rabernat committed:
add longer timeouts

rabernat commented:
By extending the timeout, we were able to move forward.

Now the build is failing because there is no fortran compiler installed.
https://app.circleci.com/jobs/github/pangeo-data/pangeo-cloud-federation/983

@cspencerjones - we have two options here
- Add a fortran compiler to the ocean environment (not sure the best way to do this)
- Create a conda-forge package for xlayers

rabernat committed:
add apt-get gfortran install

cspencerjones commented:
Looks like you're experimenting with option a)? Thanks for looking into it!

rabernat committed:
add gfortran to apt.txt [ci skip]

rabernat committed:
remove call to apt-get

rabernat committed:
update timeout to 60m

rabernat committed:
remove gfortran from apt [ci skip

rabernat committed:
install gfortran via conda

rabernat commented:
Ok, I'm giving up. I can't get xlayers to build in repo2docker via pip install.

cspencerjones commented:
Ok, so should I talk to Felipe about conda-forge?

cspencerjones committed:
Try a different conda fortran compiler

This may be a silly thing to do, but I have had success with this compiler in the past.

tjcrone commented:
@ocefpaf would be a good person to discuss this with.

ocefpaf commented:
Fortran extensions are never easy but we may help with a conda package and conda-forge's infrastructure. Looking into it now.

PS: @tjcrone, ocefpaf and Felipe (or Filipe) are the same person ;-p

tjcrone commented:
Haha yes! I was just linking to your username in case it wasn't available. Thank you Filipe! Hope you are doing well.

cspencerjones committed:
Update environment.yml

cspencerjones committed:
remove fortran compiler

cspencerjones commented:
Any chance we can merge this? Or is there something I have missed?

TomAugspurger commented:
Apologies, you'll need to merge staging again @cspencerjones to fix the merge conflict. I also included the `pip` line while fixing the dashboard issue.

cspencerjones committed:
Merge branch 'staging' into patch-2

cspencerjones commented:
Thanks @TomAugspurger . Hopefully this is ok.

cspencerjones commented:
Did I get something wrong here? Or is everyone just busy?
